src/plugins/mer/merhardwaredevicewizardpages.cpp

```diff
@@ -522,5 +529,153 @@ bool MerHardwareDeviceWizardSetupPage::validatePage()
+ * \internal
+ */
+
+class PackageKeyDeployerPrivate : public QObject
+{
+    Q_OBJECT
+public:
+    PackageKeyDeployerPrivate() = default;
+    void deploy(const SshConnectionParameters &sshParams, int index)
+    {
+        if (index < 0 || index > gpgKeyInfoList.count()) {
+            emit deployFinished(false, QString());
+            return;
+        }
+
+        const QString keyFingerprint = gpgKeyInfoList.at(index).fingerprint;
+
+        QByteArray keyData;
+        SigningUtils signUtils;
+        bool ok{false};
+        execAsynchronous(std::tie(ok, keyData), std::mem_fn(&SigningUtils::exportPublicKey), signUtils, keyFingerprint);
+        if (!ok) {
+            emit deployFinished(false, signUtils.error());
+            return;
+        }
+
+        SshRemoteProcessRunner runner;
+        QEventLoop loop;
+        connect(&runner, &SshRemoteProcessRunner::connectionError, this, [&]() {
+            emit deployFinished(false, tr("Device connection error: %1")
+                                .arg(runner.lastConnectionErrorString()));
+            loop.quit();
+        });
+        connect(&runner, &SshRemoteProcessRunner::processClosed, this, [&]() {
+            if (runner.processExitCode() == 0)
+                emit deployFinished(true, QString());
+            else
+                emit deployFinished(false, tr("Deploy key error: %1")
+                                    .arg(QString::fromUtf8(runner.readAllStandardError())));
+            loop.quit();
+        });
+        connect(&runner, &SshRemoteProcessRunner::processStarted, [&runner, &keyData]() {
+            runner.writeDataToProcess(keyData);
+        });
+        runner.run("sdk-deploy-key -", sshParams);
+        loop.exec();
+    }
+
+    GpgKeyInfoList gpgKeyInfoList;
+
+signals:
+    void deployFinished(bool ok, const QString &errorString);
+};
+
+/*!
+ * \class MerHardwareDeviceWizardPackageKeyDeploymentPage
+ * \internal
+ */
+
+MerHardwareDeviceWizardPackageKeyDeploymentPage::MerHardwareDeviceWizardPackageKeyDeploymentPage(QWidget *parent)
+    : QWizardPage(parent)
+{
+    setTitle(tr("RPM Key Deployment"));
+    setSubTitle(QLatin1String(" "));
+    p = new PackageKeyDeployerPrivate;
+
+    auto infoLabel = new QLabel;
+    infoLabel->setText(tr("Set up your device for package signature verification.\n"
+        "Depending on your device configuration, manual confirmation may be skipped for deployment of packages signed with trusted keys."));
+    infoLabel->setWordWrap(true);
+
+    m_keyComboBox = new QComboBox;
+    m_keyComboBox->setToolTip(tr("Select a trusted GnuPG public key"));
+
+    SigningUtils signUtils;
+    bool ok{false};
+    execAsynchronous(std::tie(ok, p->gpgKeyInfoList), std::mem_fn(&SigningUtils::availableKeys), signUtils);
+    if (!ok)
+        Core::MessageManager::write(signUtils.error(), Core::MessageManager::Flash);
+
+    for (const auto &keyInfo : p->gpgKeyInfoList)
+        m_keyComboBox->addItem(keyInfo.name);
```

Comment:
Here I would do similarly as I suggested for build settings: insert items in the form "NAME [FINGERPRINT]" to ensure usability in case of multiple keys for single NAME.